File: ios/ThirtySevenDegrees/DesignSystem/Organisms/AppTabBar.swift
import SwiftUI

struct AppTabBar: View {
    @Binding var selectedTab: AppTab
    var messageBadge: Int = 0

    enum AppTab: CaseIterable {
        case feed, chat, publish, discover, profile

        var title: String {
            switch self {
            case .feed: return "首页"
            case .chat: return "消息"
            case .publish: return "发布"
            case .discover: return "发现"
            case .profile: return "我的"
            }
        }

        var icon: String {
            switch self {
            case .feed: return "house.fill"
            case .chat: return "message.fill"
            case .publish: return "plus.circle.fill"
            case .discover: return "magnifyingglass"
            case .profile: return "person.fill"
            }
        }
    }

    var body: some View {
        HStack(spacing: 0) {
            ForEach(AppTab.allCases, id: \.self) { tab in
                tabButton(tab)
            }
        }
        .background(Color.bgPrimary)
        .overlay(alignment: .top) {
            Rectangle()
                .fill(Color.dividerColor)
                .frame(height: 0.5)
        }
    }

    private func tabButton(_ tab: AppTab) -> some View {
        Button {
            withAnimation(.appFast) { selectedTab = tab }
        } label: {
            VStack(spacing: 2) {
                ZStack(alignment: .topTrailing) {
                    Image(systemName: tab.icon)
                        .font(.system(size: tab == .publish ? 32 : IconSize.md.rawValue))
                        .foregroundStyle(tab == selectedTab ? Color.brandPrimary : Color.textTertiary)
                        .scaleEffect(tab == selectedTab ? 1.0 : 0.9)

                    if tab == .chat && messageBadge > 0 {
                        BadgeView(style: .count(messageBadge))
                            .offset(x: 8, y: -4)
                    }
                }
                Text(tab.title)
                    .font(.captionSmall)
                    .foregroundStyle(tab == selectedTab ? Color.brandPrimary : Color.textTertiary)
            }
            .frame(maxWidth: .infinity)
            .padding(.top, .spaceSM)
            .padding(.bottom, 4)
        }
    }
}

#Preview {
    VStack {
        Spacer()
        AppTabBar(selectedTab: .constant(.feed), messageBadge: 5)
    }
}

File: android/app/src/main/java/com/thirtysevendegrees/app/designsystem/organisms/AppTabBar.kt
package com.thirtysevendegrees.app.designsystem.organisms

import androidx.compose.animation.core.animateFloatAsState
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.vector.ImageVector
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import com.thirtysevendegrees.app.designsystem.atoms.Badge
import com.thirtysevendegrees.app.designsystem.atoms.BadgeStyle
import com.thirtysevendegrees.app.designsystem.tokens.*

enum class AppTab(
    val title: String,
    val icon: ImageVector,
    val iconSize: Dp = IconSize.MD
) {
    Feed("首页", Icons.Default.Home),
    Chat("消息", Icons.Default.Chat),
    Publish("发布", Icons.Default.AddCircle, IconSize.XL),
    Discover("发现", Icons.Default.Search),
    Profile("我的", Icons.Default.Person)
}

@Composable
fun AppTabBar(
    selectedTab: AppTab,
    onTabSelected: (AppTab) -> Unit,
    messageBadge: Int = 0,
    modifier: Modifier = Modifier
) {
    Row(
        modifier = modifier
            .fillMaxWidth()
            .background(AppColors.bgPrimaryLight),
        horizontalArrangement = Arrangement.SpaceEvenly,
        verticalAlignment = Alignment.CenterVertically
    ) {
        AppTab.entries.forEach { tab ->
            Column(
                modifier = Modifier
                    .weight(1f)
                    .clickable { onTabSelected(tab) }
                    .padding(top = Spacing.SM, bottom = 4.dp),
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                val selected = tab == selectedTab
                val scale by animateFloatAsState(
                    targetValue = if (selected) 1f else 0.9f,
                    animationSpec = AppAnimationSpec.fast(),
                    label = "tabScale"
                )

                Box {
                    Icon(
                        imageVector = tab.icon,
                        contentDescription = tab.title,
                        tint = if (selected) AppColors.brandPrimary else AppColors.textTertiaryLight,
                        modifier = Modifier
                            .size(tab.iconSize)
                            .scale(scale)
                    )
                    if (tab == AppTab.Chat && messageBadge > 0) {
                        Badge(
                            style = BadgeStyle.Count(messageBadge),
                            modifier = Modifier
                                .align(Alignment.TopEnd)
                                .offset(x = 8.dp, y = (-4).dp)
                        )
                    }
                }
                Spacer(modifier = Modifier.height(2.dp))
                Text(
                    text = tab.title,
                    style = MaterialTheme.typography.labelSmall,
                    color = if (selected) AppColors.brandPrimary else AppColors.textTertiaryLight,
                    textAlign = TextAlign.Center
                )
            }
        }
    }
}
